HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java

```diff
@@ -397,6 +400,17 @@ public int compare(final PlatformCredential obj1,
                 });
 
                 data.put("chainCertificates", chainCertificates);
+
+                if (!certificate.isBase()) {
+                    for (PlatformCredential pc : chainCertificates) {
+                        if (pc.isBase()) {
+                            if (!pc.getComponentFailures().isEmpty()) {
+                                data.put("failures", pc.getComponentFailures());
```

Comment:
Oh, taking a look at this, it'll only occur once.  There can only be one base certificate in a chain of certificates.  There for this will run once.  And the break stops the loop once the base is found.